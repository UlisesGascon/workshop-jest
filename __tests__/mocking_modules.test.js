const { getFireballs } = require('../lib/nasa_firewalls')
const { fields } = require('./fixtures/fireballs.json')
const fixtureSignature = require('./fixtures/signature')
const signatureGenerator = require('../lib/signature_generator')

// Mocks
jest.mock('../lib/signature_generator')

// Reset Mocks
beforeEach(jest.clearAllMocks)
afterEach(jest.clearAllMocks)

const validateFireballFields = fireball => {
  fields.forEach(field => {
    expect(fireball).toHaveProperty(field)
  })
}

// see: https://jestjs.io/docs/en/manual-mocks
describe('global modules mocking', () => {
  test('Mocking modules from root __mocks__ folder', async () => {
    const { data } = await getFireballs()
    expect(data).toHaveLength(10)
    data.forEach(validateFireballFields)
    expect(data).toMatchSnapshot()
  })

  test('Mocking modules manually from module __mocks__ folder', async () => {
    const { data, signature } = await getFireballs()
    expect(signature).toEqual(fixtureSignature)
    expect(signatureGenerator).toHaveBeenCalledTimes(1)
    expect(signatureGenerator).toHaveBeenCalledWith(data)
  })
})
