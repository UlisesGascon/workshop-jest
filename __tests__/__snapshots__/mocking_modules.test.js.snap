// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global modules mocking Mocking modules from root __mocks__ folder 1`] = `
Array [
  Object {
    "alt": "38.0",
    "date": "2020-08-02 16:36:25",
    "energy": "7.4",
    "impact-e": "0.23",
    "lat": "35.1",
    "lat-dir": "S",
    "lon": "34.2",
    "lon-dir": "W",
    "vel": "11.1",
  },
  Object {
    "alt": "36.0",
    "date": "2020-07-22 02:55:40",
    "energy": "2.2",
    "impact-e": "0.079",
    "lat": "20.0",
    "lat-dir": "S",
    "lon": "103.8",
    "lon-dir": "W",
    "vel": null,
  },
  Object {
    "alt": "31.6",
    "date": "2020-07-20 21:25:09",
    "energy": "5.0",
    "impact-e": "0.16",
    "lat": "51.8",
    "lat-dir": "S",
    "lon": "11.2",
    "lon-dir": "W",
    "vel": "14.1",
  },
  Object {
    "alt": "29.4",
    "date": "2020-07-12 07:50:32",
    "energy": "7.5",
    "impact-e": "0.23",
    "lat": "48.7",
    "lat-dir": "S",
    "lon": "74.0",
    "lon-dir": "E",
    "vel": "18.3",
  },
  Object {
    "alt": "29.3",
    "date": "2020-05-27 17:30:18",
    "energy": "5.7",
    "impact-e": "0.18",
    "lat": "40.8",
    "lat-dir": "N",
    "lon": "41.7",
    "lon-dir": "E",
    "vel": "14.9",
  },
  Object {
    "alt": "25",
    "date": "2020-05-18 01:11:09",
    "energy": "16.2",
    "impact-e": "0.46",
    "lat": "38.7",
    "lat-dir": "S",
    "lon": "77.2",
    "lon-dir": "E",
    "vel": null,
  },
  Object {
    "alt": null,
    "date": "2020-05-12 23:22:56",
    "energy": "8.3",
    "impact-e": "0.26",
    "lat": "15.9",
    "lat-dir": "N",
    "lon": "174.2",
    "lon-dir": "E",
    "vel": null,
  },
  Object {
    "alt": "52.5",
    "date": "2020-05-09 09:37:15",
    "energy": "4.1",
    "impact-e": "0.14",
    "lat": "7.2",
    "lat-dir": "N",
    "lon": "44.2",
    "lon-dir": "W",
    "vel": null,
  },
  Object {
    "alt": "31.2",
    "date": "2020-05-09 02:56:11",
    "energy": "4.1",
    "impact-e": "0.14",
    "lat": "44.8",
    "lat-dir": "N",
    "lon": "131.0",
    "lon-dir": "W",
    "vel": "14.5",
  },
  Object {
    "alt": null,
    "date": "2020-04-28 05:43:17",
    "energy": "2.1",
    "impact-e": "0.076",
    "lat": "20.1",
    "lat-dir": "N",
    "lon": "109.4",
    "lon-dir": "W",
    "vel": null,
  },
]
`;
